### ISSUE TITLE
[Table] prop Table.columnWidths breaks column resizing

### ISSUE THREAD

**Issue by zwerg44** [palantir/blueprint#3833]
Hello.

Looks like setting initial column width with `Table.columnWidths` breaks column resizing starting from @blueprintjs/table@3.8.1.
Version 3.8.0 - seams working.
 
#### Environment

- __Package version(s)__: @blueprintjs/table@3.8.1, @blueprintjs/core@3.19.1
- __Browser and OS versions__: firefox 70.0.1, chromium 78.0.3904.70, ubuntu 16.04

Link to a minimal repro: https://codesandbox.io/s/blueprint-sandbox-l4l11
#### Steps to reproduce

1. Create component with defined `Table.columnWidths` and resizable columns.
1. Try to resize column with mouse.

#### Actual behavior

The layout looks broken like this
![image](https://user-images.githubusercontent.com/7299028/68590626-16adf880-04a0-11ea-9c81-143069d0b568.png)

#### Possible solution

Don't have solution, I rolled back to package previous version.

**Comment by anurag060** [comment]
Did anyone find a work arround for this issue?
Hi @zwerg44 ... you got this fixed?

**Comment by zwerg44** [comment]
Hi @anurag060 
No, I rolled back versions, I still at:
    "@blueprintjs/core": "3.18.1",
    "@blueprintjs/datetime": "3.13.0",
    "@blueprintjs/select": "3.10.0",
    "@blueprintjs/table": "3.7.1",

**Comment by bheklilr** [comment]
I've also experienced this issue.  I attempted to work around it by keeping up with the column widths in a state variable and using the `onColumnWidthChanged` handler, but I happened to notice that when the table has to horizontally scroll I get a weird glitch on resize.  Any columns to the right of the one that was just resized are blanked out and completely white (pic attached).  The cells still exist and I can select them with a click.  As soon as I scroll horizontally it renders properly.  Sometimes it seems that react will just trigger a render anyway and it'll come back.

Glad to know there is a workaround to pin at 3.8.0 for now.

![image](https://user-images.githubusercontent.com/1035817/80138042-d0d67f80-8569-11ea-8ff5-bbecb61028f9.png)

**Comment by bsdis** [comment]
I also have this issue. Any plan of fixing it?

**Comment by adolfogarcia** [comment]
Was this issue ever fixed?

**Comment by alecf** [comment]
I am seeing this in 3.8.17

**Comment by aamirbutt** [comment]
I am seeing the same issue on 3.8.17. Any update about an ETA for the fix?

**Comment by lanzhenw** [comment]
I am using 3.8.26. I use `columnWidth` and  `onColumnWidthChanged` for setting default column width. I thought I would also have this issue, but it didn't break column resizing. It was working fine.

**Comment by MartinDawson** [comment]
> I am using 3.8.26. I use `columnWidth` and `onColumnWidthChanged` for setting default column width. I thought I would also have this issue, but it didn't break column resizing. It was working fine.

Please give an example because this doesn't really make sense as it isn't the correct function to use for this.

**Comment by MartinDawson** [comment]
https://github.com/palantir/blueprint/pull/3771

^ above PR broke it

3.80 works, 3.8.1 table PR broke it. Seems to be this file: https://github.com/palantir/blueprint/pull/3771/files

table.tsx change.

I'm gonna attempt to fix it now cause this bug is critical and we need to update.

**Comment by MartinDawson** [comment]
Here is the issue: https://github.com/palantir/blueprint/pull/3771/files

```js
let { columnWidths, rowHeights } = props;
        if (columnWidths == null) {
            columnWidths = state.columnWidths;
        }
        if (rowHeights == null) {
            rowHeights = state.rowHeights;
        }
```

It is never setting the columnWidths if they are supplied by props due to it being in 'controlled' mode.

```js
  const [columnWidths, setColumnWidths] = useState([200, 200, 150, 150]);
```


```js
<Table 
        columnWidths={columnWidths}
        onColumnWidthChanged={(index, width) => {
          const newColumnWidths = [...columnWidths];

          newColumnWidths[index] = width;

          setColumnWidths(newColumnWidths);
        }}

        />
```

Although there's still a major issue that the cells go white blank if you update the columnWidths and such.

**Comment by adidahiya** [comment]
this should be fixed in table v4.0.0-rc.1. if it's not, let me know and I'll re-open.
